### PR TITLE
Decompress from s3

### PR DESCRIPTION
**Pull request by mikeAdamss**
### What

Function for decompressing from s3. Minor changes from card as the tar library we're using raises errors should you try and overwrite an existing file - not having to care about the output path existing simplifies the directory/target logic significantly.

Note - If you've looked at this previously, I've also switched to using a tempfile download and the download function rather than the get object function as this enabled use of moto in the unit tests.

There is a fair bit of extra formatting included in these changes from `make fmt`. Also removed some unused imports highlighted by `make lint`.

### How to review

Sanity check the code, try it out, make sure the readme makes sense to you.

### Who can review

Anyone,.

### CODE REVIEW

File: dpytools/s3/basic.py [hunk 1]
```diff
@@ -10,9 +12,7 @@ def _get_s3_client(profile_name):
     return client
 
 
-def get_s3_object(
-    object_name: str, profile_name: Optional[str] = None
-) -> dict:
+def get_s3_object(object_name: str, profile_name: Optional[str] = None) -> dict:
     """
     Given an s3 object identifier, i.e "my-bucket/things/file.txt" returns a dictionary which
     is the boto3 aws represetation of an s3 object.
```
Comment (NickPapONS) [inline dpytools/s3/basic.py]: Should be "representation"

File: dpytools/s3/basic.py [hunk 2]
```diff
@@ -45,31 +43,23 @@ def read_s3_file_content_as_dict(
     """
     if not object_name.endswith(".json"):
         raise ValueError("Object name must end with '.json'")
```
Comment (NickPapONS) [inline dpytools/s3/basic.py:45]: Should we unit test this error? It doesn't seem significant but I don't know how much coverage we want on this.

**Review by NickPapONS — CHANGES_REQUESTED** [review]
Looks good to me, only found a typo and a question about test coverage.

File: dpytools/s3/basic.py (hunk 2 above)
Comment (mikeAdamss) [reply dpytools/s3/basic.py:45]: I think we did, this one: https://github.com/ONSdigital/dp-python-tools/blob/b6df0d428a832ab4a00baa2d56b17653fa7afba4/tests/s3/test_s3_functions.py#L63

Comment (mikeAdamss) [reply dpytools/s3/basic.py:45]: to the wider point the ideal is 100% coverage ...but.... there is a fine line between sensible coverage and wasting time on edge cases.

I trend towards cover the main branches of logic (so expected usage plus exceptions) with an eye on filling out coverage when we get to the cleaning up phase of the current tranche of work.

so yeah, we'd 100% of wanted a test for something like that if absent.

**Review by NickPapONS — APPROVED** [review]
Approved 👍

**Comment by mikeAdamss** [comment]
closing for unverified commits, code was merged elsewhere.

**Pull request closed without merging**